docs/source/_static/js/custom.js

```diff
@@ -63,3 +63,29 @@ function isValidFragment(splitFragment) {
 		}
 	}
 }());
+// Given a service name, we apply the html classes which indicate a current page to the corresponsing list item.
+// Before: <li class="toctree-l2"><a class="reference internal" href="../../acm.html">ACM</a></li>
+// After: <li class="toctree-l2 current current-page"><a class="reference internal" href="../../acm.html">ACM</a></li>
+function makeServiceLinkCurrent(serviceName) {
+	const sideBarElement = document.getElementsByClassName('sidebar-tree')[0];
+	var linkElement = sideBarElement.querySelectorAll(`a[href*="${ serviceName }"]`);
```

Comment:
Good call out. This coincidentally works since it grabs the first element it finds which is correct for all cases I see, but can easily cause issues with existing/new services who's name is a substring of another service which is listed first alphabetically (Example: `myFakeS3Service` comes before `S3`).

I think a way to be more precise is to do something like `a[href*="../${ serviceName }.html"]`. The reason for the `*=` is because a nested file can be `../../` or `../../../` depending on how deep it is in our folder structure. Another thing to narrow this search to "Available Services" section links and not just the sidebar as a whole to prevent from catching something in a different section like `guide/dynamodb.html` which we have in boto3.